packages/compiler/lib/lib.cadl

```diff
@@ -110,8 +110,8 @@ model UpdateableProperties<T> {
 }
 
 @doc("The template for omitting properties.")
-@withoutOmittedProperties(TStringOrTuple)
-model OmitProperties<T, TStringOrTuple> {
+@withoutOmittedProperties(TKeys)
+model OmitProperties<T, TKeys extends Cadl.Reflection.StringLiteral | Cadl.Reflection.Union> {
```

Comment:
I am uncomfortable using reflection types outside the context of function declarations. Users should generally think in terms of type relationships, not in terms of the syntax used to define a type. In this case, the updated definition is not significantly better than `extends string` (e.g. nothing says that the string literal has to be a key of T), and is perhaps worse in the sense that it seems to allow any union rathern than only unions which are a subtypes of string (i.e. unions with only string variants).

Reply:
I may be growing uncomfortable with reflection entirely now that I ponder further.

Reply:
Only doing this transformation at top-level, and only when it's just `string` and not `string | number`, feels confusing to me.

Reply:
wasn't just limiting that to `string` directly but any usage of string, number, boolean in a decorator